frontend/src/components/Navigator/index.js

```diff
@@ -209,19 +213,32 @@ function Navigator({
             regionId: formData.regionId,
           },
         );
-
-        // Find the goal we are editing and put it back with updated values.
-        const goalBeingEdited = allGoals.find((g) => g.name === goal.name);
-        setValue('goalForEditing', goalBeingEdited);
       }
 
+      const {
+        goals, goalForEditing: newGoalForEditing,
+      } = convertGoalsToFormData(allGoals, grantIds);
+
```

Comment:
This function unpacks the data from the API, which should now know which goal the user had open